### ISSUE TITLE
A way to automate pip-compile workflow

### ISSUE THREAD

**Issue by dspechnikov** [jazzband/pip-tools#551]
It would be awesome to have a way to automate requirements.txt recompilation. 

As far as I understand, the current workflow is:
- Write requirements.in and compile requirements.txt
- After some time, remember to manually run pip-compile again to keep dependencies fresh.

In my experience, anything I need to remember tends to be forgotten :) The first thing that comes to mind is pre-commit hook that runs pip-compile and stages updated requirements file. So at any time I'm working on the project it get latest dependencies. 

But here is another problem. Dependencies are always pinned, but I can't decide whether update them or not (maybe I don't want it right now or at all). I just blindly run pip-compile and install whatever version it tells me to install.

Something like interactive prompts from pip-review seems like a solution to this problem. I've seen the reasoning for pip-review removal and agree that we don't need the whole module for this simple task. Still, it would be great to have a tool to help decide whether to update particular dependency or not.

I'm still thinking of some git hook, but may be there are better options.

**TLDR**:
I'm thinking on implementing a solution to these problems:
- Remember to manually run pip-compile is painful, we need something builtin to automate it
- Blindly update requirements file is not always the best option, something like interactive prompt would help to decide whether it's ok or not

Any comments/suggestions?

**Comment by dspechnikov** [comment]
It turned out there is no need to additional automation since requirements.in can be used for local development without pinned requirements compilation. And CI build scripts compile them on every build anyway.
